beacon-chain/sync/service.go

```diff
@@ -173,6 +174,16 @@ func (s *Service) Stop() error {
 			s.rateLimiter.free()
 		}
 	}()
+	// Removing RPC Stream handlers.
+	for _, p := range s.p2p.Host().Mux().Protocols() {
```

Comment:
do we want to check if p2p is not nil?

Reply:
Is there any situation it would be nil ? In the event `p2p` is ever nil we wont be able to start this service up.

Reply:
was looking for its init but didn't find it